pandas/tests/arrays/test_integer.py

```diff
@@ -827,6 +827,8 @@ def test_astype_nansafe():
 
 
 @pytest.mark.parametrize("ufunc", [np.abs, np.sign])
+# np.sign emits a warning with nans, <https://github.com/numpy/numpy/issues/15127>
+@pytest.mark.filterwarnings("ignore:invalid value encountered in sign")
```

Comment:
if this warning should be suppressed, the place to do it is in the relevant code, not in the tests

Reply:
Indeed, the numpy people are working on it. See the linked bug. Until this is resolved, I don't see another way of handling it than to filter this warning here. It's very specific, too, so we're not accidentally filtering anything else.